server/service/apple_mdm.go

```diff
@@ -1514,6 +1508,21 @@ func (svc *Service) BatchSetMDMAppleProfiles(ctx context.Context, tmID *uint, tm
 		return ctxerr.Wrap(ctx, err)
 	}
 
+	if !svc.config.MDMApple.Enable {
+		// NOTE: in order to prevent an error when Fleet MDM is not enabled but no
+		// profile is provided, which can happen if a user runs `fleetctl get
+		// config` and tries to apply that YAML, as it will contain an empty/null
+		// custom_settings key, we just return a success response in this
+		// situation.
+		if len(profiles) == 0 {
+			return nil
+		}
+
+		// TODO(mna): eventually we should detect the minimum config required for
+		// this to be allowed, probably just SCEP/APNs?
+		return ctxerr.Wrap(ctx, fleet.NewInvalidArgumentError("mdm", "cannot set custom settings: Fleet MDM is not enabled"))
+	}
```

Comment:
@edwardsb This is the fix I came up with for the "MDM not enabled" issue we discussed. In addition to ignoring the validation if no profiles are sent, I moved the code around so that authorization checks are done before that case, so that any invalid authz is still raised regardless.